File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  popularityScore: number;
  weight: number;
  images: {
    yellow: string;
    rose: string;
    white: string;
  };
  price: number;
  popularityStars: number;
}

export type ColorOption = "yellow" | "rose" | "white";

export interface FilterOptions {
  minPrice?: number;
  maxPrice?: number;
  minPopularity?: number;
  maxPopularity?: number;
}

File: backend/models/product.go
package models

type Product struct {
	ID              string            `json:"id"`
	Name            string            `json:"name"`
	PopularityScore float64           `json:"popularityScore"`
	Weight          float64           `json:"weight"`
	Images          map[string]string `json:"images"`
	Price           float64           `json:"price"`
	PopularityStars float64           `json:"popularityStars"`
}

type ProductData struct {
	Name            string            `json:"name"`
	PopularityScore float64           `json:"popularityScore"`
	Weight          float64           `json:"weight"`
	Images          map[string]string `json:"images"`
}